.github/workflows/services-json.yml

```diff
@@ -0,0 +1,89 @@
+name: Services Validator
+
+on:
+  push:
+    paths:
+      - "plugins/rtmp-services/data/services.json"
+      - "plugins/rtmp-services/data/package.json"
+  pull_request:
+    paths:
+      - "plugins/rtmp-services/data/services.json"
+      - "plugins/rtmp-services/data/package.json"
+  schedule:
+  - cron: 0 0 * * *
+  workflow_dispatch:
+
+jobs:
+  schema:
+    name: Schema
+    runs-on: [ubuntu-latest]
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Install & Configure Python
+        run: |
+          sudo apt install python3.9-dev
+          python3.9 -m pip install jsonschema json_source_map
+
+      - name: Validate Service JSON Schema
+        run: |
```

Comment:
If it's a single line, you can omit the pipe - but maybe it's easier for readability to do something like:

```
JSON_FILES=(
  plugins/rtmp-services/data/services.json
  plugins/rtmp-services/data/schema/service-schema-v3.json
)
python3.9 CI/check-jsonschema.py "${JSON_FILES[@]}"
```

In that case you'd just need to add additional JSON files into the array at the top and avoid having to use backslashes to make the single line more readable.

Reply:
As an amendment, the `package.json` file should probably also be verified. Perhaps the script could be reworked to take just the to-be-verified file as input, and then load the schema from the `$schema` key and then ran in a loop over `JSON_FILES`?

e.g.
```python
try:
    with open(filename) as f:
        raw_file = f.read()
        data = json.loads(raw_file)
except OSError as e:
    print(f'Failed loading "{filename}" with {e}')
    continue

schema_filename = data.get('$schema')
if not schema_filename:
    print('File has no scehma:', filename)
    return 1

file_path = os.path.split(filename)[0]
schema_file = os.path.join(file_path, schema_filename)

try:
    with open(schema_file) as f:
        json.load(f)
except OSError as e:
    print(f'Could not load schema file: {schema_file} (Exception: {e}')
    return 1

# Schema check here
...
```